ui-tests/adminDeleteProduct.spec.js

```diff
@@ -0,0 +1,179 @@
+
+    // Handle dialog confirmation for when we click DELETE PRODUCT
+    page.once("dialog", (dialog) => {
+      console.log(`Dialog message: ${dialog.message()}`);
+      dialog.accept("yes").catch(() => {});
+    });
+    await page.getByRole("button", { name: "DELETE PRODUCT" }).click();
+
+    // Navigate back HOME
+    await page.getByText("HOME").click();
+    // Should still be on platform
+    await expect(
+      page.locator('.card-body:has-text("Test Product 2")')
+    ).toBeVisible({ timeout: 10000 });
+    // Should be deleted
+    await expect(
+      page.locator('.card-body:has-text("Test Product 1")')
+    ).not.toBeVisible({ timeout: 10000 });
+  });
+
+  test("should not delete product when dialog is not accepted with a 'yes'", async ({
+    page,
+  }) => {
+    // Check both items are visible
+    await expect(
+      page.locator('.card-body:has-text("Test Product 1")')
+    ).toBeVisible();
+
+    await expect(
+      page.locator('.card-body:has-text("Test Product 2")')
+    ).toBeVisible();
+
+    // Navigate to Product detail for cancellation of delete test
+    await page.getByRole("button", { name: "deleteproductadmin" }).click();
+    await page.getByRole("link", { name: "Dashboard" }).click();
+    await page.getByRole("link", { name: "Products" }).click();
+    await page.getByRole("link", { name: "Test Product 1" }).click();
+
+    // DELETE PRODUCT cancellation in dialog
+    page.once("dialog", (dialog) => {
+      console.log(`Dialog message: ${dialog.message()}`);
+      dialog.dismiss().catch(() => {});
+    });
+    await page.getByRole("button", { name: "DELETE PRODUCT" }).click();
+
+    // Navigate back HOME
+    await page.getByText("HOME").click();
+
+    await expect(
+      page.locator('.card-body:has-text("Test Product 2")')
+    ).toBeVisible({ timeout: 10000 });
+
+    await expect(
+      page.locator('.card-body:has-text("Test Product 1")')
+    ).toBeVisible({ timeout: 10000 });
+  });
+
+  test("should redirect to 'all products list' page after delete", async ({
+    page,
+  }) => {
+    // Check both items are visible
+    await expect(
+      page.locator('.card-body:has-text("Test Product 1")')
+    ).toBeVisible();
+
+    await expect(
+      page.locator('.card-body:has-text("Test Product 2")')
+    ).toBeVisible();
+
+    // Navigate to Product detail that is to be deleted
+    await page.getByRole("button", { name: "deleteproductadmin" }).click();
+    await page.getByRole("link", { name: "Dashboard" }).click();
+    await page.getByRole("link", { name: "Products" }).click();
+    await page.getByRole("link", { name: "Test Product 1" }).click();
+
+    // DELETE PRODUCT confirmation in dialog for redirection
+    page.once("dialog", (dialog) => {
+      console.log(`Dialog message: ${dialog.message()}`);
+      dialog.accept("yes").catch(() => {});
+    });
+    await page.getByRole("button", { name: "DELETE PRODUCT" }).click();
+
+    await expect(page).toHaveURL(
```

Comment:
Can add check to see the deleted product is not rendered

Reply:
i think to isolate test to redirection, for this test its fine as delete functionality was tested in above test cases 😬